src/common/options.cc

```diff
@@ -94,6 +157,9 @@ void Option::dump(Formatter *f) const
   f->dump_string("name", name);
 
   f->dump_string("type", type_to_str(type));
+  if (type == TYPE_SIZE) {
+    f->dump_string("units", "bytes");
```

Comment:
Seems like if we're using types to imply units, then we don't need the separate field.

Reply:
but how will this change benefit the consumers of this API in addition to more flexible `set_val_impl()` input, it can be extended to parse ``NUMBER[SUFFIX]`, where `[SUFFIX]` is `s|m|h|d` for `TYPE_SEC` settings.

Reply:
Parsing that on the input side would be great -- in the Formatter output I'm just thinking about e.g. the dashboard, which would look for type=="size_t" and then use that to apply its own bytes/kb/mb formatting -- it wouldn't ever look at a separate "units" field.

Reply:
copy! will update this PR accordingly.